Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -624,7 +624,7 @@ void ComplexTextController::advance(unsigned offset, GlyphBuffer* glyphBuffer, G
                     setHeight(paintAdvance, height(paintAdvance) - glyphOrigin(glyphIndexIntoComplexTextController + 1).y() + m_complexTextRuns[currentRunIndex + 1]->initialAdvance().height());
                 }
                 setHeight(paintAdvance, -height(paintAdvance)); // Increasing y points down
-                glyphBuffer->add(m_adjustedGlyphs[glyphIndexIntoComplexTextController], complexTextRun.font(), paintAdvance, complexTextRun.indexAt(m_glyphInCurrentRun), FloatPoint(textAutoSpaceSpacing, 0));
+                glyphBuffer->add(m_adjustedGlyphs[glyphIndexIntoComplexTextController], complexTextRun.font(), paintAdvance, complexTextRun.indexAt(m_glyphInCurrentRun) + complexTextRun.stringLocation(), FloatPoint(textAutoSpaceSpacing, 0));
```

Comment:
I think this is reasonable, I was having a look if we would need something like that on `WidthIterator` as well, but I think not. 

`ComplexTextController` splits the original string into substrings for each `ComplexTextRun` when font transition happens. Currently the offset registered in the glyphbuffer is in relation to this ComplexTextRun substring. I think it makes sense to change that as the owner of `ComplexTextController` has access to the glyph buffer and to the original string but not to the individual ComplexTextRuns